examples/complete/main.tf

```diff
@@ -1,6 +1,17 @@
 provider "aws" {
   region = "ap-south-1"
 }
+provider "aws" {
+  alias  = "cross_account_provider"
+  region = var.region
+
+  dynamic "assume_role" {
```

Comment:
let's not have this logic here, keep it simple please

Reply:
No, even with example it should be simpler. This is what others will use as reference.